src/imputors/knn.jl

```diff
@@ -73,13 +73,16 @@ function impute!(data::AbstractMatrix{Union{T, Missing}}, imp::KNN; dims=nothing
 
             # Weight valid neighbors based on inverse distance
             neighbor_dists = dists[.!neighbor_mask]
-            wv = weights(1.0 ./ neighbor_dists)
+
+            # Delay creating Weights as they don't support Inf or NaN anymore.
```

Comment:
As of #136 we can't call weights with a vector containing `Inf`, so we're just computing the sum ourselves and calling `Weights` for the weighted mean.